### PR TITLE
tracee kubernetes deployment yamls 

### PR DESCRIPTION
**Pull request by krol3**
Related the issue https://github.com/aquasecurity/tracee/issues/468
- DaemonSet for tracee
- Deployment and service for the webhook, in this sample falcosidekick
- Configmap for the trace-rules templates
- Configmap for the falcosidekick config file.

### CODE REVIEW

File: deploy/kubernetes/tracee.yaml
Comment (itaysk) [inline deploy/kubernetes/tracee.yaml:5]: can you please rename the labels according to https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels?

File: deploy/kubernetes/tracee.yaml
Comment (itaysk) [inline deploy/kubernetes/tracee.yaml:28]: Is there a reason we override the container entrypoint? I think it's better to use it and pass webhook config as additional args

File: deploy/kubernetes/tracee.yaml
Comment (itaysk) [inline deploy/kubernetes/tracee.yaml:31]: what is this port?

File: deploy/kubernetes/tracee.yaml
Comment (itaysk) [inline deploy/kubernetes/tracee.yaml:45]: we will start adding the template to the container since #683 so will not be necessary

File: deploy/kubernetes/webhook.yaml
Comment (itaysk) [inline deploy/kubernetes/webhook.yaml:5]: it's my understanding that app should be tracee in this case, as this is another component in the same app, wdyt?

File: deploy/kubernetes/webhook.yaml
Comment (itaysk) [inline deploy/kubernetes/webhook.yaml:29]: could you rename this to `tracee-webhook-config`

File: deploy/kubernetes/cfg.yaml
Comment (itaysk) [inline deploy/kubernetes/cfg.yaml:1]: can we make this a proper kubernetes configmap resource? (`kind: ConfigMap`)

File: deploy/kubernetes/Readme.md
Comment (itaysk) [inline deploy/kubernetes/Readme.md:8]: ideally, after my suggestions below, this entire example could be boiled down to `kubectl apply -f ./deploy`

File: deploy/kubernetes/Readme.md
Comment (itaysk) [inline deploy/kubernetes/Readme.md:6]: Can you add a note that 

> This approach assumes that kernel headers are available on the Kubernetes nodes under conventional location, e.g `/usr/src` and `/lib/modules`

Comment (itaysk) [reply deploy/kubernetes/Readme.md:6]: after merging this we should spend some time improving this experience: instructions to build the eBPF probe once using the tracee image, and then use the slim image in the daemonset. for a first step this is fine

**Review by itaysk — CHANGES_REQUESTED** [review]

File: deploy/kubernetes/Readme.md
Comment (krol3) [reply deploy/kubernetes/Readme.md:8]: in the future we could have deploy/helm, similar folder starboard struct. Do you want to remove the kubernetes folder ?

File: deploy/kubernetes/webhook-cm.yaml
Comment (itaysk) [inline deploy/kubernetes/webhook-cm.yaml:11]: let's comment these out so it won't be shipped in a broken state

File: deploy/kubernetes/tracee.yaml
Comment (itaysk) [inline deploy/kubernetes/tracee.yaml:10]: I don't think manged-by means that, it means which tool manages this. we can remove it

File: deploy/kubernetes/tracee.yaml
Comment (itaysk) [inline deploy/kubernetes/tracee.yaml:7]: I'm not sure how to manage the version with static yamls. also the image ref is latest, so let's remove it

File: deploy/kubernetes/tracee.yaml
Comment (itaysk) [inline deploy/kubernetes/tracee.yaml:6]: this instance label is meaningless if we put the same value in all instances, so let's remove it

File: deploy/kubernetes/Readme.md
Comment (itaysk) [inline deploy/kubernetes/Readme.md:1]: can you please move this to the docs under "installation"?

File: deploy/kubernetes/Readme.md
Comment (itaysk) [inline deploy/kubernetes/Readme.md:23]: I think this is all unnecessary after the latest changes right?
The only thing I would keep is the mention of falcosidekick, and a call out for the user to change the configmap before deploying

File: deploy/kubernetes/webhook-cm.yaml
Comment (itaysk) [inline deploy/kubernetes/webhook-cm.yaml:6]: would be nice to add a comment here pointing to the reference: https://github.com/falcosecurity/falcosidekick/blob/master/config_example.yaml

**Review by itaysk — APPROVED** [review]
LGTM! (assuming you tested it on an actual k8s?)
I just added one small suggestion on the webhook configmap

**Comment by krol3** [comment]
It was testing in the KUBE_VERSION=1.20.5

**Comment by simar7** [comment]
Just revisiting this, how do I use the provided deployment yaml files? This is what I did:

```
vagrant@ubuntu-groovy:~/tracee$ k apply -f deploy/kubernetes/tracee.yaml 
daemonset.apps/tracee created

vagrant@ubuntu-groovy:~/tracee$ k get pods
NAME                        READY   STATUS              RESTARTS   AGE
tracee-qsgt2                0/1     Running   0          2s

vagrant@ubuntu-groovy:~/tracee$ k logs tracee-qsgt2
Loaded signature(s):  [TRC-1 TRC-2 TRC-3 TRC-4 TRC-5 TRC-6 TRC-7]
2021/06/11 02:20:45 kernel is not properly configured, missing: [CONFIG_BPF CONFIG_BPF_SYSCALL CONFIG_KPROBE_EVENTS CONFIG_BPF_EVENTS]
```


I understand the requirements and do have them on my host machine: https://github.com/aquasecurity/tracee/blob/main/docs/install/kubernetes.md#deploy-on-kubernetes – I'm curious how do they translate into the k8s environment? I assume if a pod is running in privileged mode, it should have access to the appropriate headers, etc. correct?

**Pull request merged**
